prboom2/src/p_inter.c

```diff
@@ -187,43 +190,47 @@ static dboolean P_GiveAmmo(player_t *player, ammotype_t ammo, int num)
     {
```

Comment:
You can put `if (!autoswitch) return true;` before `if (heretic)` and remove these 2 conditions

Reply:
Ah great catch! Also simplifies this section much more nicely :)